File: csharp/1346.check-if-n-and-its-double-exist.cs
/*
 * @lc app=leetcode id=1346 lang=csharp
 *
 * [1346] Check If N and Its Double Exist
 */

using System.Collections.Generic;

// @lc code=start
public class Solution {
    public bool CheckIfExist(int[] arr) {
        var set = new HashSet<int>();
        foreach (int x in arr) {
            if (set.Contains(x * 2) || x % 2 == 0 && set.Contains(x / 2)) {
                return true;
            }
            set.Add(x);
        }
        return false;
    }
}
// @lc code=end

File: java/1346.check-if-n-and-its-double-exist.java
/*
 * @lc app=leetcode id=1346 lang=java
 *
 * [1346] Check If N and Its Double Exist
 */

import java.util.HashSet;
import java.util.Set;

// @lc code=start

class Solution {
    public boolean checkIfExist(int[] arr) {
        Set<Integer> seen = new HashSet<>();
        for (int num : arr) {
            if (num % 2 == 0 && seen.contains(num / 2)) {
                return true;
            }
            if (seen.contains(num * 2)) {
                return true;
            }
            seen.add(num);
        }
        return false;
    }
}
// @lc code=end

